src/app/clusters/ethernet_network_diagnostics_server/ethernet_network_diagnostics_server.cpp

```diff
@@ -16,11 +16,106 @@
  */
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterceptor.h>
 #include <app/CommandHandler.h>
+#include <app/MessageDef/AttributeDataElement.h>
 #include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/Optional.h>
+#include <platform/ConnectivityManager.h>
 
 using namespace chip;
+using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EthernetNetworkDiagnostics::Attributes;
+using chip::DeviceLayer::ConnectivityManager;
+
+namespace {
+
+class EthernetDiagosticsAttrInterceptor : public AttributeAccessInterceptor
+{
+public:
+    // Register for the EthernetNetworkDiagnostics cluster on all endpoints.
+    EthernetDiagosticsAttrInterceptor() :
+        AttributeAccessInterceptor(Optional<EndpointId>::Missing(), EthernetNetworkDiagnostics::Id)
+    {}
+
+    CHIP_ERROR ReadAttributeData(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter, bool * aDataRead) override;
+
+    bool IsRegistered() const { return sIsRegistered; }
+
+    void RegisterIfNeeded()
```

Comment:
These shouldn't really be _public_ methods. Perhaps it would be cleaner to shift this logic into `emberAfEthernetNetworkDiagnosticsClusterServerInitCallback` since it's really just meant be invoked in the context of initialization.

```
void emberAfEthernetNetworkDiagnosticsClusterServerInitCallback(EndpointId endpoint)
{
    static bool isRegistered = false;

    if (!isRegistered) {
          gAttrInterceptor.RegisterIfNeeded();
          isRegistered = true;
    }
}
```

Reply:
> Perhaps it would be cleaner to shift this logic into emberAfEthernetNetworkDiagnosticsClusterServerInitCallback

That's a fair point.  I can do that.